string name
int32 x
int32 y